### PR TITLE
replace remaining usage of sh with subprocess_util

### PR DESCRIPTION
**Pull request by gotmax23**

### CODE REVIEW

File: src/antsibull_core/ansible_core.py [hunk 1]
```diff
@@ -252,43 +252,31 @@ async def checkout_from_git(download_dir: str, repo_url: str = _ANSIBLE_CORE_URL
...
     ansible_core_dir = os.path.join(download_dir, 'ansible-core')
-    await loop.run_in_executor(None, sh.git, 'clone', repo_url, ansible_core_dir)
+    await async_log_run(['git', 'clone', repo_url, ansible_core_dir])
 
     return ansible_core_dir
 
 
-@lru_cache(None)
+@cache
 async def create_sdist(source_dir: str, dest_dir: str) -> str:
```
Comment (gotmax23) [inline src/antsibull_core/ansible_core.py:262]: This function isn't used anywhere (antsibull has its own code to build sdist and wheels). IMO, we should remove it in antsibull-core v2. What do you think?

Comment (felixfontein) [reply src/antsibull_core/ansible_core.py:262]: I'm fine with removing it (as long as that's documented, and as long as the integration tests with the older antsibull versions do not fail - I don't think they should, but who knows :shrug:).

**Comment by felixfontein** [comment]
BTW, I started something similar for antsibull-docs, but noticed I can do essentially nothing until https://github.com/ansible-community/antsibull-docs/pull/120 and #50 are done.

File: src/antsibull_core/collections.py [hunk 2]
```diff
@@ -22,24 +19,18 @@ class CollectionFormatError(Exception):
...
-
-        # If the choice of install tools for galaxy is ever settled upon, we can switch from tar to
-        # using that
-        # sh dynamically creates functions which map to executables
-        # pyre-ignore[16] pylint:disable-next=no-member
-        installers.append(loop.run_in_executor(executor, sh.tar, '-xf', pathname, '-C',
-                                               collection_dir))
+        installers.append(
+            asyncio.create_task(
+                async_log_run(['-xf', pathname, '-C', collection_dir])
```
Comment (felixfontein) [inline src/antsibull_core/collections.py:31]: This is missing the actual command. (I guess that would be `tar`?)

File: src/antsibull_core/collections.py [hunk 3]
```diff
@@ -79,12 +66,11 @@ async def install_separately(collection_tarballs: list[str], collection_dir: str
...
 
-        # If the choice of install tools for galaxy is ever settled upon, we can switch from tar to
-        # using that
-        # sh dynamically creates functions which map to executables
-        # pyre-ignore[16] pylint:disable-next=no-member
-        installers.append(loop.run_in_executor(executor, sh.tar, '-xf', pathname, '-C',
-                                               collection_dir))
+        installers.append(
+            asyncio.create_task(
+                async_log_run(['-xf', pathname, '-C', collection_dir])
```
Comment (felixfontein) [inline src/antsibull_core/collections.py:71]: Same here.

**Comment by gotmax23** [comment]
(Sorry in advanced for the force pushes. I'd normally run the integration tests locally but am relying on CI given my bad connection.)

File: src/antsibull_core/ansible_core.py (hunk 1 above)
Comment (gotmax23) [reply src/antsibull_core/ansible_core.py:262]: Doh, this is used later in this same file...

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@gotmax23 thanks for working on this!

**Pull request merged**